src/service/ticket.ts

```diff
@@ -0,0 +1,53 @@
+import { FastifyReply, FastifyRequest } from "fastify";
+import { Ticket } from "../entities";
+
+/**
+ * Retrieves a ticket from DB. It returns error either;
+ * 1. Passed ticket id is invalid - not convertible to Number
+ * 2. Ticket couldn't be found by the passed id
+ */
+export async function getTicketById(ticketId: string): Promise<Ticket> {
+  const ticketIdQuery = Number(ticketId);
+
+  if (isNaN(ticketIdQuery) || !ticketIdQuery) throw new Error(`Invalid ticket id: ${ticketIdQuery}`);
+
+  const ticket = await Ticket.findOneBy({
+    id: ticketIdQuery,
+  });
+
+  if (!ticket) throw new Error(`Ticket not found for id: ${ticketIdQuery}`);
+
+  return ticket;
+}
+
+export async function updateTicket(originalTicket: Ticket, updateRequestTicket: Partial<Ticket>) {
+  // Check if there's any unexpected key included in the request
+  const isAllRequestKeyAssignable = Object.keys(updateRequestTicket).some((key) => {
```

Comment:
`isAllRequestKeysAssignable`?
sorry if I am wrong

Reply:
Oh doesn't the singleton rule apply here? I can change it to `Keys` too

Reply:
Fixed!